app/api/tax.py

```diff
@@ -28,92 +29,94 @@ def before_post(self, args, kwargs, data):
         require_relationship(['event'], data)
         if not has_access('is_coorganizer', event_id=data['event']):
             raise ForbiddenException({'source': ''}, 'Co-organizer access is required.')
+        if get_count(db.session.query(Event).filter_by(id=int(data['event']), is_tax_enabled=False)) > 0:
+            raise MethodNotAllowed({'parameter': 'event_id'}, "Tax is disabled for this Event")
 
-    def before_get(self, args, kwargs):
+    def before_create_object(self, data, view_kwargs):
         """
-        method to assign proper schema based on admin access
-        :param args:
-        :param kwargs:
+        method to check if tax object already exists for an event
+        :param data:
+        :param view_kwargs:
         :return:
         """
-        if 'Authorization' in request.headers and has_access('is_admin'):
-            self.schema = TaxSchema
+        try:
+            self.session.query(Tax).filter_by(event_id=data['event']).one()
+        except NoResultFound:
```

Comment:
Can we use the following instead of `try-except-else`:
```
if not  self.session.query(Tax).filter_by(event_id=data['event']).first():
    raise ConflictException({'pointer': '/data/relationships/event'}, "Tax already exists for this event")
```

Please correct me if this is not appropriate 😄

Reply:
It will cause HTTP 500 IMO. 
```self.session.query(Tax).filter_by(event_id=data['event'])``` can be None and when you apply ```first``` on it, it will raise HTTP 500.

Reply:
No, not here but it raised No Result Found in Stripe Authorization.
PR: https://github.com/fossasia/open-event-server/pull/4867

Reply:
@srv-twry Okay, so if possible, can you please test it and see whether `NoResultFound` is raised? Because I used a similar statement in my PR #4851  and tested it, but the error was not raised, as you know :)

Reply:
Lessons learnt:
1. ```.first()``` returns None when the data is not found but ```.one``` raises `NoResultFound` exception.
2. Never copy paste code provided by reviewers :laughing: 
`if not self.session.query(Tax).filter_by(event_id=data['event']).first():` this will raise exception when tax is not defined. We just need the opposite.